### PR TITLE
Upgrade publishing workflows

### PR DESCRIPTION
**Pull request by t0yv0**
Upgrade publishing workflows to more modern versions borrowing from pulumi/pulumi-aws. Fixes Node SDK publishing. Since Node SDK is now generated under sdk/node and is no longer special compared to other SDKs some changes were needed to the GitHub Actions publishing process to get it to work right.

### CODE REVIEW

**Comment by t0yv0** [comment]
Artifact uploaded. I downloaded it and checked it, it seems to have the package.json present. 

```
{
    "name": "@pulumi/eks",
    "version": "2.9.0-alpha.1727283921+0e06def",
    "keywords": [
        "pulumi",
        "aws",
        "eks"
    ],
    "homepage": "https://pulumi.com",
    "repository": "https://github.com/pulumi/pulumi-eks",
    "license": "Apache-2.0",
    "scripts": {
        "build": "tsc"
    },
    "dependencies": {
        "@pulumi/aws": "^6.18.2",
        "@pulumi/kubernetes": "^4.1.1",
        "@pulumi/pulumi": "^3.47.0",
        "https-proxy-agent": "^5.0.1",
        "jest": "^29.7.0",
        "js-yaml": "^4.1.0",
        "netmask": "^2.0.2",
        "semver": "^7.3.7",
        "which": "^1.3.1"
    },
    "devDependencies": {
        "@types/js-yaml": "^4.0.5",
        "@types/netmask": "^1.0.30",
        "@types/node": "^18.11.13",
        "@types/semver": "^7.3.10",
        "@types/which": "^1.3.1",
        "typescript": "^4.6.2"
    },
    "pulumi": {
        "resource": true,
        "name": "eks",
        "version": "2.9.0-alpha.1727283921+0e06def"
    }
}
```

I think this fixes https://github.com/pulumi/pulumi-eks/issues/1385 but to prove it out we need to pubish an alpha end to end and try it I think.

**Comment by flostadler** [comment]
The script for publishing also has some of the old paths hardcoded: https://github.com/pulumi/pulumi-eks/blob/d4f539f99700f3cd084da85f3dad59ec018a2dc0/scripts/publish_packages.sh#L12-L17

**Comment by t0yv0** [comment]
Ack, looking

**Comment by t0yv0** [comment]
I've checked with @guineveresaenger and am trying to adapt pulumi-aws publish.yml here for this repo to bring it closer to state of the art. This needs to be tested since we're not sure. I'll be tagging some commits here with alpha tags to test getting a pre-release out all the way.

**Comment by t0yv0** [comment]
Tagged v3.0.0-alpha.4 but that didn't quite work. Will use a temporary setup to run master.yml on this branch also.

**Comment by t0yv0** [comment]
I've iterated on running master.yml manually..

Last run succeeded publishing this:

https://www.npmjs.com/package/@pulumi/eks?activeTab=versions

[2.9.0-alpha.1727304793](https://www.npmjs.com/package/@pulumi/eks/v/2.9.0-alpha.1727304793)

**Comment by t0yv0** [comment]
It failed to publish PyPy auto version https://github.com/pulumi/pulumi-eks/actions/runs/11042365693

**Comment by t0yv0** [comment]
I can tag a commit and try again. This might help test it on a version that's more like an alpha version. Also I can possibly make release.yml testable and test a pre-release from this branch.

File: .github/actions/setup-tools/action.yml [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    default: all
+
+runs:
+  using: "composite"
+  steps:
+    - name: Install Go
+      if: inputs.tools == 'all' || contains(inputs.tools, 'go')
+      uses: actions/setup-go@v5
+      with:
+        go-version: "1.23.x"
```
Comment (flostadler) [inline .github/actions/setup-tools/action.yml:25]: Nice! Just checked, eks was using 1.21 before

File: .github/actions/setup-tools/action.yml [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+      if: inputs.tools == 'all' || contains(inputs.tools, 'schema-tools')
+      uses: jaxxstorm/action-install-gh-release@v1.11.0
+      with:
+        repo: pulumi/schema-tools
+
+    - name: Setup Node
+      if: inputs.tools == 'all' || contains(inputs.tools, 'nodejs')
+      uses: actions/setup-node@v4
+      with:
+        node-version: 20.x
```
Comment (flostadler) [inline .github/actions/setup-tools/action.yml:54]: Can we use the `.node-version` file here? We were using node 18 before. Ideally the major version upgrade shouldn't cause issues here, but I'm a bit worried about upgrading too many things at once.

WDYT about doing that separately?

File: .github/workflows/master.yml [hunk 3]
```diff
@@ -203,154 +193,38 @@ jobs:
...
+    uses: ./.github/workflows/publish.yml
+    secrets: inherit
+    with:
+      version: ${{ needs.prerequisites.outputs.version }}
+      isPrerelease: true
+      skipGoSdk: true
+
+  publish_test:
+    name: publish_test
+    if: (github.head_ref || github.ref_name) == 't0yv0/fix-compressing-node-sdk'
```
Comment (flostadler) [inline .github/workflows/master.yml:205]: Let's not forget to remove the testing setup before merging

**Review by flostadler — APPROVED** [review]
Nice, the changes look good to me! Just a bit worried about changing too many things at once (go & node version).

**Pull request merged**